monai/data/grid_dataset.py

```diff
@@ -44,7 +48,7 @@ def __init__(self, dataset, patch_size, start_pos=(), mode: str = "wrap", **pad_
         self.dataset = dataset
         self.patch_size = (None,) + tuple(patch_size)
         self.start_pos = start_pos
-        self.mode = mode
+        self.mode = NumpyPadMode(mode)
```

Comment:
`self.mode: NumpyPadMode = NumpyPadMode(mode)`
To explicitly type self.mode.  While the __init__ may be a string or a NumpyPadMode, internally only NumpyPadMode is vaild.

Reply:
```python
self.mode = NumpyPadMode(mode)
reveal_type(self.mode)
```

```bash
$> mypy monai/data/grid_dataset.py 
monai/data/grid_dataset.py:52: note: Revealed type is 'monai.utils.enums.NumpyPadMode*'
```

Internally mypy already places that restriction (the asterisk denotes the type was inferred).

Being explicit would serve to protect the type by having mypy throw an error if the ``NumpyPadMode()`` were to be removed.

Reply:
@benjamin-gorman I am a code safety, explicit is better than implicit, junkie.  I would prefer that all "revealed types" were explicit.